Python/challenges/day_4.py

```diff
@@ -0,0 +1,31 @@
+def check_for_containment_pairs(pairs: list, range_overlap=False):
+    pair1 = pairs[0].split("-")
+    pair2 = pairs[1].split("-")
+    if int(pair1[0]) == int(pair2[0]) or int(pair1[1]) == int(pair2[1]):
+        return True  
```

Comment:
This expression is redundant, because it's fully covered in the next two expressions. It is true though that if one boundary is the same for both pairs, that one pair must be fully contained by the other pair.

You don't have to remove this line, this is just extra info.